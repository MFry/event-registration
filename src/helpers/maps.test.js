import googleMapsUrl from "./maps";

describe("Getting a valid url from data", () => {
  it("Returns a correct url with ", () => {
    const address = {
      street: "621 W 6th Ave",
      city: "Anchorage",
      state: "Alaska",
      zipcode: "99501"
    };
    expect(googleMapsUrl(address)).toMatchSnapshot();
  });
  it("Returns the correct url without zipcode", () => {
    const address = {
      street: "5000 W International Airport Rd",
      city: "Anchorage",
      state: "Alaska"
    };
    expect(googleMapsUrl(address)).toMatchSnapshot();
  });
  it("Returns the correct url with extra white space", () => {
    const address = {
      street: "621 W 6th Ave",
      city: "Anchorage",
      state: "Alaska",
      zipcode: "99501"
    };
    const malformedAddress = {
      street: "\t621   W 6th   Ave\t",
      city: "Anchorage\t\t",
      state: "Alaska",
      zipcode: "99501"
    };
    expect(googleMapsUrl(address)).toEqual(googleMapsUrl(malformedAddress));
  });
});
