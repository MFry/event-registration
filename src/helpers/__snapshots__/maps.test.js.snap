// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting a valid url from data Returns a correct url with  1`] = `"https://www.google.com/maps/dir/?api=1&destination=621+W+6th+Ave+Anchorage+Alaska"`;

exports[`Getting a valid url from data Returns the correct url without zipcode 1`] = `"https://www.google.com/maps/dir/?api=1&destination=5000+W+International+Airport+Rd+Anchorage+Alaska"`;
